docs/ai/snippets/microsoft-extensions-ai/AI.Shared/SampleEmbeddingGenerator.cs

```diff
@@ -4,32 +4,29 @@ public sealed class SampleEmbeddingGenerator(
     Uri endpoint, string modelId)
         : IEmbeddingGenerator<string, Embedding<float>>
 {
-    public EmbeddingGeneratorMetadata Metadata { get; } =
-        new(nameof(SampleEmbeddingGenerator), endpoint, modelId);
+    private readonly EmbeddingGeneratorMetadata _metadata =
+        new("SampleEmbeddingGenerator", endpoint, modelId);
 
     public async Task<GeneratedEmbeddings<Embedding<float>>> GenerateAsync(
         IEnumerable<string> values,
         EmbeddingGenerationOptions? options = null,
         CancellationToken cancellationToken = default)
     {
-        // Simulate some async operation
+        // Simulate some async operation.
         await Task.Delay(100, cancellationToken);
 
-        // Create random embeddings
-        return
-        [
-            .. from value in values
+        // Create random embeddings.
+        return new GeneratedEmbeddings<Embedding<float>>(
+            from value in values
             select new Embedding<float>(
-                Enumerable.Range(0, 384)
-                          .Select(_ => Random.Shared.NextSingle())
-                          .ToArray())
-        ];
+                Enumerable.Range(0, 384).Select(_ => Random.Shared.NextSingle()).ToArray()));
     }
 
-    public object? GetService(Type serviceType, object? serviceKey) => this;
-
-    public TService? GetService<TService>(object? key = null)
-        where TService : class => this as TService;
+    public object? GetService(Type serviceType, object? serviceKey) =>
+        serviceKey is not null ? null :
+        serviceType == typeof(EmbeddingGeneratorMetadata) ? _metadata :
+        serviceType?.IsInstanceOfType(this) is true ? this :
+        null;
```

Comment:
This code is hard to read in my opinion. The ternary operators are easier to read per line:

```cs
some true|false expression
    ? true branch
    : false branch